django/core/management/commands/migrate.py

```diff
@@ -164,10 +164,27 @@ def handle(self, *args, **options):
         if options['prune']:
             if not options['app_label']:
                 raise CommandError('Migrations can be pruned only when an app is specified.')
-            to_prune = set(executor.loader.applied_migrations) - set(executor.loader.disk_migrations)
-            to_prune = sorted(migration for migration in to_prune if migration[0] == app_label)
             if self.verbosity > 0:
                 self.stdout.write('Pruning migrations:', self.style.MIGRATE_HEADING)
+            to_prune = set(executor.loader.applied_migrations) - set(executor.loader.disk_migrations)
+            squashed_migrations_with_deleted_replaced_migrations = [
+                migration_key for migration_key, migration_obj in executor.loader.replacements.items() if any(
+                    replaced in to_prune for replaced in migration_obj.replaces
+                )
+            ]
+            if squashed_migrations_with_deleted_replaced_migrations:
+                to_prune = []
+                msg = (
+                    '  Pruning cannot take place until the following squashed '
+                    "migrations are recorded as applied (re-run 'manage.py migrate') "
+                    'and have their replaces attribute removed:'
+                )
+                self.stdout.write(msg, self.style.NOTICE)
+                for migration_to_warn in squashed_migrations_with_deleted_replaced_migrations:
```

Comment:
I guess I could wrap this in a `sorted()` in case the insertion order of `loader.replacements` ever turns out a little funny.